library/func_misc.py

```diff
@@ -153,3 +153,55 @@ def Call(self, pos_args, named_args):
 
         raise error.TypeErr(x, 'len() expected Str, List, or Dict',
                             loc.Missing)
+
+
+class Join(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        r = typed_args.Reader(pos_args, named_args)
+        li = r.PosList()
+
+        delim = r.NamedStr('delim', '')
+        if len(pos_args): # reader has a reference
+            delim = r.PosStr()
+
+        r.Done()
+
+        strs = []  # type: List[str]
+        for i, el in enumerate(li):
+            msg = 'join() items must be Str but item %d is %s' % (
+                i, value_str(el.tag()))
+            strs.append(val_ops.ToStr(el, msg, loc.Missing))
```

Comment:
what do you think about val_ops.py.Stringify() ?  It is used for splicing like `@array` and also `$str`


I guess that doesn't match what Python or JS do, so it's up for debate

Reply:
Stringifying would actually be pretty nice. It's very jarring when I'm using python interactively and `','.join(foo)`  barfs because some member of `foo` is an int or something making you have to type out a whole `map(...)` to serialize things. For an interactive shell, I think it's appropriate to have some magic here